site/src/modules/resources/AgentLogs.tsx

```diff
@@ -0,0 +1,276 @@
+import Tooltip from "@mui/material/Tooltip";
+import { FixedSizeList as List } from "react-window";
+import type { WorkspaceAgentLogSource } from "api/typesGenerated";
+import {
+  LogLine,
+  logLineHeight,
+} from "modules/workspaces/WorkspaceBuildLogs/Logs";
+import {
+  ComponentProps,
+  forwardRef,
+  useEffect,
+  useMemo,
+  useRef,
+  useState,
+} from "react";
+import { LineWithID } from "./AgentRow";
+import { Interpolation, Theme } from "@emotion/react";
+import * as API from "api/api";
+
+type AgentLogsProps = Omit<
+  ComponentProps<typeof List>,
+  "children" | "itemSize" | "itemCount"
+> & {
+  logs: LineWithID[];
+  sources: WorkspaceAgentLogSource[];
+};
+
+export const AgentLogs = forwardRef<List, AgentLogsProps>(
+  ({ logs, sources, ...listProps }, ref) => {
+    const logSourceByID = useMemo(() => {
+      const sourcesById: { [id: string]: WorkspaceAgentLogSource } = {};
+      for (const source of sources) {
+        sourcesById[source.id] = source;
+      }
+      return sourcesById;
+    }, [sources]);
```

Comment:
> do we know how big the dataset is expected to get? 
It can be a lot. It depends on the user's use case. It can have 3+ scripts, and each script can have more than +200  log lines for example.